### PR TITLE
Fix check mode in iptables_state for incomplete iptables-save files along with integration tests

### PR DESCRIPTION
**Pull request by Maxopoly**
##### SUMMARY

Fixes https://github.com/ansible-collections/community.general/issues/7463

##### ISSUE TYPE

- Bugfix Pull Request
- Test Pull Request

##### COMPONENT NAME

iptables_state

##### ADDITIONAL INFORMATION

This PR first attempted to reproduce the behavior observed in https://github.com/ansible-collections/community.general/issues/7463 through integration tests. iptables-save doesn't work in docker, so this was tested against the azure pipeline through this PR, please excuse the commit spam.

https://github.com/ansible-collections/community.general/commit/a58bc16f2fb9ef4e5e60346abad4474db6d4933a was able reproduce the issue in https://github.com/ansible-collections/community.general/issues/7463 based on incomplete iptables-save files. Initially, all tables are empty and uninitialized, which will make iptables-save return nothing for them.

For example when creating a nat rule and then deleting it, an output of iptables-save may look like this:

```
        "*filter",
        ":INPUT ACCEPT [0:0]",
        ":FORWARD DROP [0:0]",
        ":OUTPUT ACCEPT [0:0]",
        "COMMIT",
        "*nat",
        ":PREROUTING ACCEPT [0:0]",
        ":INPUT ACCEPT [0:0]",
        ":OUTPUT ACCEPT [0:0]",
        ":POSTROUTING ACCEPT [0:0]",
        "COMMIT",
```
while one on a new container/machine which has never had a nat rule will look like this:

```
        "*filter",
        ":INPUT ACCEPT [0:0]",
        ":FORWARD DROP [0:0]",
        ":OUTPUT ACCEPT [0:0]",
        "COMMIT",
```

This broke the existing check_mode checks, which did simple string comparison. Most likely there were also edge cases outside of check mode in regards to uninitialized tables of existing iptables-save dumps, which would report changes when restoring them, despite nothing changing. This is only the case for iptables-save files created through means other than this module.

To circument this I implemented parsing of iptables-save dumps into a simple data structure based on preexisting code.  If a table currently has some kind of state (including rules) but is not referenced in the iptables-save dump, it will not be touched by iptables-restore and now handled properly by this module.

I also added the tables `tables_after` after restoring as a data structure, similar to the already existing state `tables` before running the module

### CODE REVIEW

File: changelogs/fragments/8029-iptables-state-restore-check-mode.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - iptables_state - Fix idempotency issues when restoring incomplete iptables dumps and add target state as data structure to output (https://github.com/ansible-collections/community.general/issues/8029).
```
Comment (felixfontein) [inline changelogs/fragments/8029-iptables-state-restore-check-mode.yml:2]:
```suggestion
  - iptables_state - fix idempotency issues when restoring incomplete iptables dumps and add target state as data structure to output (https://github.com/ansible-collections/community.general/issues/8029).
```

File: plugins/modules/iptables_state.py [hunk 2]
```diff
@@ -233,6 +235,36 @@
       ]
     }
   returned: always
+tables_after:
+  description:
+    - The iptables on the system after the module has run, separated by table.
```
Comment (felixfontein) [inline plugins/modules/iptables_state.py:240]: This is a new feature. Would you mind moving this to a separate PR so the bugfix can be backported? (Or alternatively move the bugfix to a separate PR.)

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by Maxopoly** [comment]
Fixed as suggested and removed the additional return value introduced. I'll make an additional PR against master to readd that once this PR is merged, as it depends on the changes made here.

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@Maxopoly thanks for your contribution!
